proposal/custom-url-resolution.md

```diff
@@ -0,0 +1,230 @@
+# Custom url resolution: Draft 1
+
+_[(Issue)](https://github.com/sass/sass/issues/2535)_
+
+## Table of Contents
+
+- [Background](#background)
+- [Summary](#summary)
+  - [Example](#example)
+- [Function](#function)
+  - [Steps](#steps)
+  - [JavaScript API](#javascript-api)
+- [Presets](#presets)
+  - [The inline Preset](#the-inline-preset)
+    - [Inline Preset Usage](#inline-preset-usage)
+    - [Inline Preset Implementation](#inline-preset-implementation)
+  - [The relative Preset](#the-relative-preset)
+    - [Relative Preset Usage](#relative-preset-usage)
+    - [Relative Preset Implementation](#relative-preset-implementation)
+
+## Background
+
+> This section is non-normative.
+
+Many css features require the use of a sass-url reference to reference resources from outside the sass files, however these files also need to exist on the eventual output directory and server. To ensure the references are valid, the sass API should allow for the user to provide a way to remap and/or inline these resources.
+
+## Summary
+
+> This section is non-normative.
+
+This proposal defines a standardized way to remap the sass-url references to the final location on the server or inline reference.
+
+This is accomplished by running sass-url references through the url rewriting plugin if a url rewriting plugin has been defined.
+
+### Example
+
+This example demonstrates the possibility of rewriting urls to inline assets using data URIs.
+
+Input:
+
+```scss
+.upload-icon {
+  background-url: sass-url('../icons/upload-icon.svg');
+}
+```
+
+Output:
+
+```css
+.upload-icon {
+  background-url: url('data:image/svg+xml;base64,Jggg==');
+}
+```
+
+## Function
+
+This proposal introduces logic for a new function: `sass-url("...")`, this function allow users to define a url that gets rewritten based on the config to ensure it can be loaded correctly from the browser.
```

Comment:
This is the sort of thing that would be better in the summary. It does introduce the concept of a `sass-url` function (that's good!), but it's mostly just a nonspecific explanation of its general purpose. It's also not clear what "the config" refers to here.